Lib/typing.py

```diff
@@ -273,10 +274,24 @@ def _collect_parameters(args):
                         parameters.append(collected)
         elif hasattr(t, '__typing_subst__'):
             if t not in parameters:
+                if t.__default__ is not None:
```

Comment:
Not sure we can assume the attribute is always present. What if the user uses `typing_extensions.ParamSpec` from an old version of typing_extensions, for example?